python/cuda_parallel/resources/cmake/FindCython.cmake

```
@@ -0,0 +1,88 @@
+#.rst:
+#
+# Find ``cython`` executable.
+#
+# This module will set the following variables in your project:
+#
+#  ``CYTHON_EXECUTABLE``
+#    path to the ``cython`` program
+#
+#  ``CYTHON_VERSION``
+#    version of ``cython``
+#
+#  ``CYTHON_FOUND``
+#    true if the program was found
+#
+# For more information on the Cython project, see https://cython.org/.
+#
+# *Cython is a language that makes writing C extensions for the Python language
+# as easy as Python itself.*
+#
+#=============================================================================
+# Copyright 2011 Kitware, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+# Use the Cython executable that lives next to the Python executable
+# if it is a local installation.
+if(Python_EXECUTABLE)
+  get_filename_component(_python_path ${Python_EXECUTABLE} PATH)
+elseif(Python3_EXECUTABLE)
+  get_filename_component(_python_path ${Python3_EXECUTABLE} PATH)
+elseif(DEFINED PYTHON_EXECUTABLE)
+  get_filename_component(_python_path ${PYTHON_EXECUTABLE} PATH)
+endif()
```

Consider calling `find_package(Python3 COMPONENTS Interpreter)` here to just have to check `Python3_EXECUTABLE`.

In either case, the paths should be quoted:

```suggestion
if(Python_EXECUTABLE)
  get_filename_component(_python_path "${Python_EXECUTABLE}" PATH)
elseif(Python3_EXECUTABLE)
  get_filename_component(_python_path "${Python3_EXECUTABLE}" PATH)
elseif(DEFINED PYTHON_EXECUTABLE)
  get_filename_component(_python_path "${PYTHON_EXECUTABLE}" PATH)
endif()
```